plugins/suspendmanager.cpp

```diff
@@ -291,6 +298,70 @@ class SuspendManager {
         return false;
     }
 
+    static bool suitableToMoveItemTo(df::coord pos) {
+        if (!walkable(pos))
+            return false;
+
+        // Dwarves will move a blocking item to a walkable building, but only if its fully constructed
+        auto building = Buildings::findAtTile(pos);
+        return !building || building->flags.bits.exists;
+    }
+
+    // True for buildings planned on a tile with an item that cannot be moved aside
+    static bool isOnUnmovableItem(df::job *job) {
+        CHECK_NULL_POINTER(job);
+        auto building = Job::getHolder(job);
+        if (!building)
+            return false;
+
+        // Check for items on the building, look into all the map blocks overlapping
+        // the building
+        bool has_item = false;
+        for (int32_t block_x = building->x1 >> 4; block_x <= building->x2 >> 4; ++block_x) {
+            for (int32_t block_y = building->y1 >> 4; block_y <= building->y2 >> 4; ++block_y) {
+                auto block = Maps::getBlock(block_x,block_y,building->z);
+                if (!block)
+                    continue;
+
+                for (df::item *item : block->items | transform(df::item::find)) {
+                    if (!item)
+                        continue;
+                    if (item->pos.x < building->x1 || item->pos.x > building->x2 ||
+                        item->pos.y < building->y1 || item->pos.y > building->y2)
+                        continue;
+
+                    // Ok if it's an item of the job
+                    if (std::ranges::any_of(job->items, [item](df::job_item_ref *j) {return j->item == item;}))
+                        continue;
+                    if (item->flags.bits.in_job || item->flags.bits.forbid)
+                        return true; // Assigned to a different task or forbidden, not movable
+                    has_item = true;
+                }
+            }
+        }
+
+        if (!has_item)
+            return false;
+
+        // With an item, check for the surrounding of the building
+        // The position of the item does not matter, in multi-tile buildings
+        // dwarves will move the item multiple tiles if necessary
+        for (auto x : {building->x1-1, building->x2+1}) {
+            for (auto y = building->y1; y <= building->y2; ++y) {
+                if (suitableToMoveItemTo(coord(x,y,building->z)))
+                    return false;
+            }
+        }
+        for (auto y : {building->y1-1, building->y2+1}) {
+            for (auto x = building->x1; x <= building->x2; ++x) {
+                if (suitableToMoveItemTo(coord(x,y,building->z)))
+                    return false;
+            }
+        }
```

Comment:
are the corners valid positions to move items to? They're not included in this check.

Reply:
Nope, the corners are not valid in my tests, I'm updating the comment.

Here the axle is getting suspended by the dwarves for as long as the wall is not built. Without wall, they will transport the annoying log all the way up though:

![image](https://github.com/DFHack/dfhack/assets/630159/07c64436-1d2d-4e1e-9449-feee17f2abe6)